libcaf_openssl/caf/openssl/ssl_session.hpp

```diff
@@ -0,0 +1,59 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright (C) 2011 - 2017                                                  *
+ * Dominik Charousset <dominik.charousset (at) haw-hamburg.de>                *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#ifndef CAF_OPENSSL_SSL_SESSION_HPP
+#define CAF_OPENSSL_SSL_SESSION_HPP
+
+#include "caf/actor_system.hpp"
+#include "caf/io/network/native_socket.hpp"
+
+#include <openssl/err.h>
```

Comment:
Doesn't really work well, as the types are typedefs to internal structs. I've removed the err.h include though, don't need that in the header.